plugins/woocommerce-admin/client/core-profiler/pages/Plugins.tsx

```diff
@@ -0,0 +1,190 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf, _n } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import interpolateComponents from '@automattic/interpolate-components';
+import { Link } from '@woocommerce/components';
+import { Extension, ExtensionList } from '@woocommerce/data';
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { CoreProfilerStateMachineContext } from '../index';
+import { PluginsInstallationRequestedEvent, PluginsPageSkippedEvent } from '..';
+import { Heading } from '../components/heading/heading';
+import { Navigation } from '../components/navigation/navigation';
+import { PluginCard } from '../components/plugin-card/plugin-card';
+import { getAdminSetting } from '~/utils/admin-settings';
+
+const locale = getAdminSetting( 'locale' ).siteLocale.replace( '_', '-' );
+const joinWithAnd = ( items: string[] ) => {
+	return new Intl.ListFormat( locale, {
+		style: 'long',
+		type: 'conjunction',
+	} ).format( items );
+};
+
+export const Plugins = ( {
+	context,
+	navigationProgress,
+	sendEvent,
+}: {
+	context: CoreProfilerStateMachineContext;
+	sendEvent: (
+		payload: PluginsInstallationRequestedEvent | PluginsPageSkippedEvent
+	) => void;
+	navigationProgress: number;
+} ) => {
+	const [ selectedPlugins, setSelectedPlugins ] = useState<
+		ExtensionList[ 'plugins' ]
+	>( context.pluginsAvailable.filter( ( plugin ) => ! plugin.is_installed ) );
+
+	const setSelectedPlugin = ( plugin: Extension ) => {
+		setSelectedPlugins(
+			selectedPlugins.some( ( item ) => item.key === plugin.key )
+				? selectedPlugins.filter( ( item ) => item.key !== plugin.key )
+				: [ ...selectedPlugins, plugin ]
+		);
+	};
+
+	const skipPluginsPage = () => {
+		return sendEvent( {
+			type: 'PLUGINS_PAGE_SKIPPED',
+		} );
+	};
+	const submitInstallationRequest = () => {
+		return sendEvent( {
+			type: 'PLUGINS_INSTALLATION_REQUESTED',
+			payload: {
+				plugins: selectedPlugins.map( ( plugin ) =>
+					plugin.key.replace( ':alt', '' )
+				),
+			},
+		} );
+	};
+	const errorMessage = context.pluginsInstallationErrors.length
+		? interpolateComponents( {
+				mixedString: sprintf(
+					// Translators: %s is a list of plugins that does not need to be translated
+					__(
+						'Oops! We encountered a problem while installing %s. {{link}}Please try again{{/link}}.',
+						'woocommerce'
+					),
+					joinWithAnd(
+						context.pluginsInstallationErrors.map(
+							( error ) => error.plugin
+						)
+					)
+				),
+				components: {
+					link: (
+						<Button isLink onClick={ submitInstallationRequest } />
+					),
+				},
+		  } )
+		: null;
+
+	const pluginsWithAgreement = selectedPlugins.filter(
+		( plugin ) =>
+			plugin.key === 'jetpack' ||
+			plugin.key === 'woocommerce-services:shipping' ||
+			plugin.key === 'woocommerce-services:tax'
```

Comment:
A minor suggestion: We could use `.includes` here instead.

```ts
	const pluginsWithAgreement = selectedPlugins.filter( ( plugin ) =>
		[
			'jetpack',
			'woocommerce-services:shipping',
			'woocommerce-services:tax',
		].includes( plugin.key )
	);
```

Reply:
Thanks for the suggestion 👍 Updated in https://github.com/woocommerce/woocommerce/pull/38405/commits/14c41fdf88996a093444d7bdf5272d599aed09ef